src/borg/_hashindex.c

```diff
@@ -520,9 +672,25 @@ hashindex_write(HashIndex *index, PyObject *file_py)
             /* Be able to work with regular file objects which do not have a hash_part method. */
             PyErr_Clear();
         } else {
-            return;
+            return 0;
         }
     }
+    return 1;
+}
+
+#ifndef BORG_NO_PYTHON
+static void
+hashindex_write(HashIndex *index, PyObject *file_py, int legacy)
+{
+    PyObject *length_object, *buckets_view;
+    Py_ssize_t length;
+    Py_ssize_t buckets_length = (Py_ssize_t)index->num_buckets * index->bucket_size;
+
+    if (legacy)
+        1 / 0;  // guess we do not write old hashindexes
```

Comment:
What is this about?

Reply:
old hashindex is only used when reading from borg 1 repos.

borg 2 users might do that once per repo to transfer old archives to new borg 2 repos.
but as the old repo is only read, writing the old hashindex is not used.

in case my assumption were wrong, i just wanted it to blow up there.